src/Apps/Order/Routes/Shipping2/AddressModal2.tsx

```diff
@@ -95,7 +95,9 @@ export const AddressModal: React.FC<Props> = ({
     null
   )
   const [showDialog, setShowDialog] = useState<boolean>(false)
-  const { shipsFrom } = useShippingContext()
+  const {
+    savedOrderData: { shipsFrom },
+  } = useShippingContext()
```

Comment:
It is helpful for understanding origins to avoid destructuring everywhere. It might feel cleaner and more fun to do this, but in complex code -- in the future -- its important to know where things originate from. Ie, 

```
const shippingContext = useShippingContext()
...
if (shippingContext.savedOrderData.shipsFrom) {
  ...
}
```
yields an enormous amount of contextual info. It says: data comes from context; data is from savedOrderData; and it is the shipsFrom field. Where `shipsFrom` alone says nothing at a glance. The more complex the code, the more important immediately-at-a-glance metadata like this is becomes, even if it might not feel that way as the author.